src/libraries/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/SymmetricAlgorithm.cs

```diff
@@ -428,6 +428,220 @@ public int GetCiphertextLengthCfb(int plaintextLength, PaddingMode paddingMode =
             }
         }
 
+        /// <summary>
+        ///   Decrypts data using ECB mode with the specified padding mode.
+        /// </summary>
+        /// <param name="ciphertext">The data to decrypt</param>
+        /// <param name="paddingMode">The padding mode used to produce the ciphertext and remove during decryption</param>
+        /// <returns>The decrypted plaintext data</returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="ciphertext" /> is <see langword="null" />.
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="paddingMode" /> is not a valid padding mode.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <see cref="TryDecryptEcbCore" /> could not decrypt the ciphertext.
+        /// </exception>
+        /// <remarks>
+        ///   This method's behavior is defined by <see cref="TryDecryptEcbCore" />.
+        /// </remarks>
+        public byte[] DecryptEcb(byte[] ciphertext, PaddingMode paddingMode)
+        {
+            // Padding mode is validated by callee.
+            if (ciphertext is null)
+                throw new ArgumentNullException(nameof(ciphertext));
+
+            return DecryptEcb(new ReadOnlySpan<byte>(ciphertext), paddingMode);
+        }
+
+        /// <summary>
+        ///   Decrypts data using ECB mode with the specified padding mode.
+        /// </summary>
+        /// <param name="ciphertext">The data to decrypt</param>
+        /// <param name="paddingMode">The padding mode used to produce the ciphertext and remove during decryption</param>
+        /// <returns>The decrypted plaintext data.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="paddingMode" /> is not a valid padding mode.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <see cref="TryDecryptEcbCore" /> could not decrypt the ciphertext.
+        /// </exception>
+        /// <remarks>
+        ///   This method's behavior is defined by <see cref="TryDecryptEcbCore" />.
+        /// </remarks>
+        public byte[] DecryptEcb(ReadOnlySpan<byte> ciphertext, PaddingMode paddingMode)
+        {
+            CheckPaddingMode(paddingMode);
+
+            // This could get returned directly to the caller if we there was no padding
+            // that needed to get removed, so don't rent from a pool.
+            byte[] decryptBuffer = GC.AllocateUninitializedArray<byte>(ciphertext.Length);
```

Comment:
For things other than ECB padding is expected.  For ECB "no padding" might be the more appropriate expectation (being a building block for some other mode).

So maybe this is fine.  Maybe it'd be nicer to make it be this when padding mode is zero, and a rent otherwise.

Reply:
I took a stab at this and adding renting made the code uncomfortable and not worth the complexity, I think. I happier to go all-renting or all-allocating.

> For things other than ECB padding is expected.

Since this is in ECB mode specifically my personal thought is this is okay, but when CBC/CFB>8 come around I agree using `ArrayPool` would make the most sense.